### PR TITLE
Show warning when filename start/end with whitespace in explorer inputbox

### LINKED ISSUE #91187: Should not allow file or directory names to start with space
I realize this is entirely user error, but I just wasted a good amount of time trying to track down why a path wasn't resolving, only to find that when creating a folder I had accidentally added a space to the beginning of the name 🤦‍♂ 

Screen Shot 2020-02-21 at 1 52 57 PM

I can't think of a case where this would actually be desired? It would be great if VS Code wouldn't allow this, and show a validation error if a file or folder name starts with a space. 

I'd be happy to attempt to open a PR for this if this is something you would accept and if someone could point me to the right place to start in the codebase.

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #91187

@isidorn I end up adding a new optional attribute `notificationMessage` to the `IEditableData` interface. Let me know what you think.

### CODE REVIEW

**Comment by isidorn** [comment]
@jeanp413 thanks for this PR. While this approach makes sense I would like to minimize changes if possible.
I do not see a clear need for a new method `notificationMessage`, why not just reuse and enhance `validationMessage`? I think that will make it cleaner and leaner.

**Comment by jeanp413** [comment]
I tried that at first but it seemed like more changes at that time 😅 
If I update the validationMessage signature to
 `validationMessage: (value: string) => { content: string, severity: Severity } | null;`
I'll need to make some changes in `tunnelView.ts`

Another option could be ` string | { content: string, severity: Severity } | null` to avoid touching `tunnelView.ts`

Thoughts?

**Comment by isidorn** [comment]
I think making changes to the `tunnelView` is perfectly fine, if the changes are for the better.
@alexr00 fyi

**Comment by jeanp413** [comment]
Pushed some changes updating `validationMessage` return type

File: src/vs/workbench/contrib/remote/browser/tunnelView.ts [hunk 1]
```diff
@@ -733,7 +733,17 @@ namespace ForwardPortAction {
 						}
 						remoteExplorerService.setEditable(undefined, null);
 					},
-					validationMessage: validateInput,
+					validationMessage: (value) => {
```
Comment (alexr00) [inline src/vs/workbench/contrib/remote/browser/tunnelView.ts]: Thanks for making the changes to tunnelView.ts! The code in both `validationMessage` here is duplicated. Can you refactor into a function in this file:
```typescript
function validationMessage(validationString: string | null): { content: string, severity: Severity } | null {
	if (!validationString) {
		return null;
	}

	return {
		content: validationString,
		severity: Severity.Error
	};
}
```
Then in the two `validationMessage` you just need to have:
```typescript
					validationMessage: (value) => {
						return validationMessage(validateInput(value));
					},
```

**Review by alexr00 — CHANGES_REQUESTED** [review]

**Review by alexr00 — APPROVED** [review]
The tunnelView changes look good!

**Comment by isidorn** [comment]
Works great, awesome work!
Merging in.

**Pull request merged**